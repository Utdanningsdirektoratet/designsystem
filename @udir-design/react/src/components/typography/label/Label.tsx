import {
  Label as DigdirLabel,
  type LabelProps as DigdirLabelProps,
} from '@digdir/designsystemet-react';
import { ComponentRef, ForwardRefExoticComponent, RefAttributes } from 'react';

type LabelProps = Omit<DigdirLabelProps, 'data-color'>;

const Label = DigdirLabel as ForwardRefExoticComponent<
  LabelProps & RefAttributes<ComponentRef<typeof DigdirLabel>>
>;

export { Label, LabelProps };
