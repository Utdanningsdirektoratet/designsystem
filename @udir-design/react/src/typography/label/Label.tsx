export { Label, type LabelProps } from '@digdir/designsystemet-react';
