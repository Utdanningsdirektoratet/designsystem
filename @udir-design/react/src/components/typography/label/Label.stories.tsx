import type { Meta, StoryObj } from '@storybook/react-vite';
import { Label } from './Label';

const meta: Meta<typeof Label> = {
  component: Label,
  tags: ['beta', '!autodocs', 'digdir'],
  parameters: {
    componentOrigin: {
      originator: 'digdir',
    },
  },
  title: 'Components/Typography/Label',
};

export default meta;
type Story = StoryObj<typeof Label>;

export const Preview: Story = {
  args: {
    children: 'Vennligst skriv inn f√∏dselsnummer. 11 tegn',
    weight: 'medium',
  },
};
