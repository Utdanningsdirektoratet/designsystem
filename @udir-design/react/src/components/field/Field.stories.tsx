import type { Meta, StoryObj } from '@storybook/react-vite';
import {
  Divider,
  Field,
  Input,
  Textarea,
  ValidationMessage,
} from '@udir-design/react/alpha';
import { Label } from '../typography/label/Label';

const meta: Meta<typeof Field> = {
  component: Field,
  tags: ['alpha'],
  parameters: {
    customStyles: {
      maxWidth: 600,
      width: '90vw',
    },
    layout: 'centered',
  },
};

export default meta;
type Story = StoryObj<typeof Field>;

export const Preview: Story = {
  render: () => (
    <Field>
      <Label>Etternavn</Label>
      <Field.Description>
        Etternavn kan ikke inneholde mellomrom
      </Field.Description>
      <Input id="preview" defaultValue="Nordmann Svenske" />
      <ValidationMessage>
        Du kan ikke ha mellomrom i etternavnet ditt
      </ValidationMessage>
    </Field>
  ),
};

export const Affix: Story = {
  render: () => (
    <Field>
      <Label>Hvor mange kroner koster det per måned?</Label>
      <Field.Affixes>
        <Field.Affix>NOK</Field.Affix>
        <Input id="affix" />
        <Field.Affix>pr. mnd.</Field.Affix>
      </Field.Affixes>
    </Field>
  ),
};

export const Counter: Story = {
  render: () => (
    <Field>
      <Label>Legg til en beskrivelse</Label>
      <Textarea rows={2} id="counter" />
      <Field.Counter limit={10} />
    </Field>
  ),
};

export const Position: Story = () => (
  <>
    <Field position="end">
      <Label>Flymodus</Label>
      <Input type="checkbox" role="switch" id={'airplane'} />
    </Field>
    <Divider />
    <Field position="end">
      <Label>Lydløs</Label>
      <Input type="checkbox" role="switch" id={'sounds'} />
    </Field>
  </>
);

Position.decorators = [
  (Story) => (
    <div
      style={{
        maxWidth: 200,
        margin: 'auto',
      }}
    >
      <Story />
    </div>
  ),
];
