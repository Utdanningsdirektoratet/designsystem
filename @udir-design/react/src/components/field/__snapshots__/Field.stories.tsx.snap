// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Affix 1`] = `
<div>
  <label
    data-weight="medium"
    for="affix"
  >
    Hvor mange kroner koster det per måned?
  </label>
  <div>
    <span aria-hidden="true">
      NOK
    </span>
    <input
      id="affix"
      type="text"
    >
    <span aria-hidden="true">
      pr. mnd.
    </span>
  </div>
</div>
`;

exports[`Counter 1`] = `
<div>
  <label
    data-weight="medium"
    for="counter"
  >
    Legg til en beskrivelse
  </label>
  <textarea
    rows="2"
    id="counter"
    aria-describedby="counter:validation:1  counter:description:1"
  >
  </textarea>
  <div
    data-field="description"
    aria-live="polite"
    id="counter:description:1"
  >
  </div>
  <p
    data-variant="default"
    data-field="validation"
    id="counter:validation:1"
  >
    10 tegn igjen
  </p>
</div>
`;

exports[`Position 1`] = `
<div style="max-width: 200px; margin: auto;">
  <div data-position="end">
    <label
      data-weight="medium"
      for="airplane"
    >
      Flymodus
    </label>
    <input
      role="switch"
      id="airplane"
      type="checkbox"
    >
  </div>
  <hr aria-hidden="true">
  <div data-position="end">
    <label
      data-weight="medium"
      for="sounds"
    >
      Lydløs
    </label>
    <input
      role="switch"
      id="sounds"
      type="checkbox"
    >
  </div>
</div>
`;

exports[`Preview 1`] = `
<div>
  <label
    data-weight="medium"
    for="preview"
  >
    Etternavn
  </label>
  <div
    data-field="description"
    id="preview:description:1"
  >
    Etternavn kan ikke inneholde mellomrom
  </div>
  <input
    id="preview"
    type="text"
    value="Nordmann Svenske"
    aria-describedby="preview:validation:1  preview:description:1"
  >
  <p
    data-field="validation"
    id="preview:validation:1"
  >
    Du kan ikke ha mellomrom i etternavnet ditt
  </p>
</div>
`;
